### PR TITLE
Edit docs and README for docs migration to Splunk

### PR DESCRIPTION
**Pull request by theletterf**
The official Java OTel docs are now on our docs site: 

https://docs.splunk.com/Observability/gdi/get-data-in/application/java/get-started.html

I've pushed a small MR to update our repo README to point to the official docs when appropriate, as per our new content strategy. Docs will still remain here though, to facilitate knowledge transfer.

### CODE REVIEW

**Review by mateuszrzeszutek — APPROVED** [review]
Nice 👍

File: README.md
Comment (iNikem) [inline README.md:37]: How do we make sure that this list is updated when we add something to this repo?

File: README.md
Comment (iNikem) [inline README.md:177]: Why this was removed? Also I am somewhat concerned about having documentation in `docs` folder which is not linked from anywhere.

**Review by iNikem — COMMENTED** [review]
@flands Would like to have your approval as well.

File: README.md
Comment (theletterf) [reply README.md:37]: We update it here and I'm mentioned. As per CODEOWNERS, I should stay abreast of all changes in the .md files here.

File: README.md
Comment (theletterf) [reply README.md:177]: This is a specific feature/use case that is a bit outside of the scope of the README—we either put them all or none.

But we can reinstate it, with a link to the official docs if needed.

I understand the concern about the docs, but O11y docs should be the primary source that we link. We can use local docs for knowledge transfer while we think about automation.

Comment (mateuszrzeszutek) [reply README.md:177]: Can we leave a link to official docs in the README? Now that I think of it, there's a requirement in the GDI spec about that one specific use case:

```md
* MUST document how to configure log correlation
```

Comment (theletterf) [reply README.md:177]: I'll add it in my next commit.

Comment (theletterf) [reply README.md:177]: Fixed!

**Review by mateuszrzeszutek — APPROVED** [review]

File: README.md
Comment (flands) [inline README.md:103]: I would like to propose we keep this -- if someone directly navigates to repo or someone wants to contribute to this repo, this is useful information (I understand it means duplication).

**Review by flands — COMMENTED** [review]
LGTM -- I would like to propose one change

**Review by iNikem — APPROVED** [review]
Pending Steve's comment about getting started...

File: README.md
Comment (theletterf) [reply README.md]: Yes, excellent idea. We achieve two things by keeping instructions there:

- We keep a place where the team can update the smoothest path to success
- We ensure that, should the Splunk docs be down, people can still set it up

I've brought them back in https://github.com/signalfx/splunk-otel-java/pull/373/commits/38294c3b62c982166ff4d032291d3e6022183512, taken from the current instructions that I've improved.

File: README.md
Comment (mateuszrzeszutek) [inline README.md:124]:
```suggestion
    java -javaagent:./splunk-otel-javaagent.jar \
      -jar <myapp>.jar
```

File: README.md
Comment (mateuszrzeszutek) [inline README.md:127]: If we're quoting `-javaagent` we probably should do the same for `-jar`

```suggestion
    Insert the `-javaagent` flag before the `-jar` file, adding it as a JVM option, not as an application argument.
```

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by flands — APPROVED** [review]
LGTM

**Pull request merged**
